### PR TITLE
Kelsey/announcements2

### PR DESCRIPTION
**Pull request by charlesx2013**

### CODE REVIEW

File: sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementActivity.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    private int mType;
+    public static final int ORIGINAL = 0;
+    public static final int DELETED = 1;
+    public static final int EDITED = 2;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState){
+        super.onCreate(savedInstanceState);
+        Announcement announcement = null;
+        mType = ORIGINAL;
```
Comment (kelseylam) [inline sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementActivity.java]: mType is originally 0 anyways so its k

File: sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementActivity.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            announcement = mapper.readValue(getIntent().getStringExtra("Announcement"), Announcement.class);
+            FragUtils.replace(R.id.container, AnnouncementFragment.newInstance(announcement), this);
+
+        } catch (IOException exception) {
+            String error = "Announcement";
+            Toast.makeText(this, getString(R.string.cannot_be_displayed, error), Toast.LENGTH_SHORT).show();
```
Comment (kelseylam) [inline sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementActivity.java:40]: you can pass in a resource directly, no need to call getString

File: sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementActivity.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    public void setType(int type) {
+        mType = type;
+    }
+
+    @Override
+    public void finish() {
+        Intent intent = new Intent();
+        Bundle bundle = new Bundle();
+        String string;
+        if (mType == DELETED) {
```
Comment (kelseylam) [inline sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementActivity.java]: pls use switch

File: sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementFormAbstractFragment.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            if (mAnnouncement.getSchool() != null) {
+                setSchoolSpinner();
+            }
+        }
+    }
+
+    public abstract void initializeViews();
+
+    public void initializeSpinners() {
+        String[] categoryList = new String[]{SCHOOL, GENERAL};
```
Comment (kelseylam) [inline sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementFormAbstractFragment.java]: refactor into class variable

File: sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementFormAbstractFragment.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+        StringArraySpinnerAdapter categoryAdapter = new StringArraySpinnerAdapter(getActivity(), categoryList, R.layout.simple_spinner_item, R.layout.simple_spinner_item);
+        categoryAdapter.setDropDownViewResource(android.R.layout.simple_spinner_item);
```
Comment (kelseylam) [inline sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementFormAbstractFragment.java]: delete this - we set dropdown view resource on our own above

File: sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementFormAbstractFragment.java [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+        mAnnouncementCategory.setAdapter(categoryAdapter);
+        mAnnouncementCategory.setSelection(1, true);
+        mSchoolAdapter = new SchoolSpinnerAdapter(getActivity(), mSchools, R.layout.simple_spinner_item, R.layout.simple_spinner_item);
+        mSchoolAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
```
Comment (kelseylam) [inline sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementFormAbstractFragment.java]: same

File: sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementFragment.java [hunk 7]
```diff
@@ -44,21 +59,78 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.menu_edit:
+                FragUtils.replaceBackStack(R.id.container, EditAnnouncementFragment.newInstance(mAnnouncement), getActivity());
+                break;
+            case R.id.menu_delete:
+                Requests.Announcements.with(getActivity()).makeDeleteRequest(mAnnouncement);
```
Comment (kelseylam) [inline sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementFragment.java]: add confirm dialog

File: sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementFragment.java [hunk 8]
```diff
@@ -44,21 +59,78 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+        mBody.setText(mAnnouncement.getBody());
     }
 
     public void setAnnouncement(Announcement announcement) {
-        this.announcement = announcement;
+        mAnnouncement = announcement;
     }
+
+    @Override
+    public void onStart() {
```
Comment (kelseylam) [inline sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementFragment.java]: style: move overriden methods to the top

File: sage-android/Sage/app/src/main/java/blueprint/com/sage/main/fragments/AnnouncementsListFragment.java [hunk 9]
```diff
@@ -85,11 +100,52 @@ public void initializeViews() {
...
+        String string = data.getStringExtra(getString(R.string.create_announcement));
+        Announcement announcement = NetworkUtils.writeAsObject(getActivity(), string, new TypeReference<Announcement>(){});
+        mAnnouncements.add(0, announcement);
+        mAdapter.setAnnouncements(mAnnouncements);
+        mAdapter.notifyDataSetChanged();
+    }
+
+    public void changeAnnouncement(Intent data) {
+        int type = data.getIntExtra(getString(R.string.announcement_type), AnnouncementActivity.ORIGINAL);
+        if (type == AnnouncementActivity.DELETED) {
```
Comment (kelseylam) [inline sage-android/Sage/app/src/main/java/blueprint/com/sage/main/fragments/AnnouncementsListFragment.java]: switch pls

Comment (kelseylam) [inline sage-android/Sage/app/src/main/java/blueprint/com/sage/main/fragments/AnnouncementsListFragment.java]: staement

File: sage-android/Sage/app/src/main/java/blueprint/com/sage/main/fragments/AnnouncementsListFragment.java [hunk 10]
```diff
@@ -85,11 +100,52 @@ public void initializeViews() {
...
+            mAdapter.setAnnouncements(mAnnouncements);
+            mAdapter.notifyDataSetChanged();
+        } else if (type == AnnouncementActivity.EDITED) {
+            String string = data.getStringExtra(getString(R.string.edit_announcement));
+            Announcement announcement = NetworkUtils.writeAsObject(getActivity(), string, new TypeReference<Announcement>() {});
+            for (int i = 0; i < mAnnouncements.size(); i++) {
+                if (mAnnouncements.get(i).getId() == announcement.getId()) {
+                    mAnnouncements.set(i, announcement);
+                }
+            }
```
Comment (kelseylam) [inline sage-android/Sage/app/src/main/java/blueprint/com/sage/main/fragments/AnnouncementsListFragment.java:138]: repetitve code - pls refactor

**Comment by charlesx2013** [comment]
Yay!

**Comment by charlesx2013** [comment]
LGTM

**Pull request merged**